File: src/main/java/net/finance/bo/SavingsBo.java
package net.finance.bo;

import lombok.NonNull;
import net.finance.entity.Savings;
import net.finance.repository.SavingsRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;
import java.math.BigDecimal;

@Service
@Transactional
public class SavingsBo {

    @NonNull
    private final SavingsRepository savingsRepository;

    @Autowired
    public SavingsBo(SavingsRepository savingsRep) {
        savingsRepository = savingsRep;
    }

    public Savings create(Savings Savings) {
        return savingsRepository.save(Savings);
    }

    public void delete(Integer id) {
        savingsRepository.deleteById(id);
    }

    public Savings get(Integer id) {
        return savingsRepository.findById(id).get();
    }

    public Page<Savings> list(PageRequest pageReq) {
        return savingsRepository.findAll(pageReq);
    }

    public Savings update(Savings dev) {
        return savingsRepository.save(dev);
    }

    public BigDecimal getTotalSavings() {
        return savingsRepository.getSumSavings();
    }
}

File: src/main/kotlin/com/finances/bo/SavingsBo.kt
package com.finances.bo

import com.finances.dto.SavingsDTO
import com.finances.entity.Savings
import com.finances.mapper.toDTO
import com.finances.repository.SavingsRepository
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageRequest
import org.springframework.stereotype.Service
import java.math.BigDecimal

@Service
class SavingsBo(private val savingsRepository: SavingsRepository) {

    fun create(savings: Savings): SavingsDTO {
        return savingsRepository.save(savings).toDTO()
    }

    fun delete(id: Int) {
        savingsRepository.deleteById(id)
    }

    fun get(id: Int): SavingsDTO {
        return savingsRepository.findById(id).orElseThrow { NoSuchElementException() }.toDTO()
    }

    fun list(pageReq: PageRequest): Page<SavingsDTO> {
        return savingsRepository.findAll(pageReq).map(Savings::toDTO)
    }

    fun update(dev: Savings): SavingsDTO {
        return savingsRepository.save(dev).toDTO()
    }

    val totalSavings: BigDecimal
        get() = savingsRepository.sumSavings()
}
